tt_metal/tt_metal.cpp

```diff
@@ -797,10 +800,19 @@ void CompileProgram(Device *device, Program &program, bool fd_bootloader_mode) {
 }
 
 void AllocateBuffer(Buffer *buffer, bool bottom_up) {
+    if(GraphTracker::instance().hook_allocate(buffer, bottom_up)) {
+        GraphTracker::instance().track_allocate(buffer, bottom_up);
+        return;
+    }
     EnqueueAllocateBuffer(buffer->device()->command_queue(), buffer, bottom_up, false);
+    GraphTracker::instance().track_allocate(buffer, bottom_up);
 }
```

Comment:
As discussed, just give it a thought. 
I see a benefit of splitting tracking and hooks.
But it also seem possible to implement Tracking via hooks.

```suggestion
void AllocateBuffer(Buffer *buffer, bool bottom_up) {
    if(GraphTracker::instance().hook_allocate(buffer, bottom_up)) {
        return;
    }
    EnqueueAllocateBuffer(buffer->device()->command_queue(), buffer, bottom_up, false);
}
```